jdaviz/configs/imviz/plugins/catalogs/catalogs.py

```diff
@@ -159,6 +159,8 @@ def search(self, error_on_fail=False):
                     zoom_radius = r_max
                 query_region_result = SDSS.query_region(skycoord_center, radius=zoom_radius,
                                                         data_release=17)
+                if len(query_region_result) > self.max_sources:
+                    query_region_result = query_region_result[:self.max_sources]
```

Comment:
can we have a warning for this (maybe in-UI)?

Reply:
I added a snackbar message instead of a warning since the user manually enters the `max_sources` themselves. Also added it for the Gaia catalog with the custom logic for this.